EncodingStreamingHighAvailability/HighAvailability/Services/MediaServiceInstanceHealthService.cs

```diff
@@ -207,22 +221,40 @@ public async Task<IEnumerable<MediaServiceInstanceHealthModel>> ReEvaluateMediaS
                         // if duration below max threshold, it is "healthy" in progress, otherwise it is counted as "unheatlhy"
                         if (duration.TotalMinutes < this.numberOfMinutesInProcessToMarkJobStuck)
                         {
-                            inHealthyProgressCount++;
+                            jobsInHealthyProgressCount++;
                         }
                     }
                 }
 
-                logger.LogInformation($"MediaServiceInstanceHealthService::ReEvaluateMediaServicesHealthAsync aggregated data: instanceName={mediaServiceInstanceHealthModel.MediaServiceAccountName} successCount={successCount} totalCount={totalCount} inHealthyProgressCount={inHealthyProgressCount}");
+                logger.LogInformation($"MediaServiceInstanceHealthService::ReEvaluateMediaServicesHealthAsync jobs aggregated data: instanceName={mediaServiceInstanceHealthModel.MediaServiceAccountName} jobSuccessCount={jobsSuccessCount} jobTotalCount={jobsTotalCount} jobInHealthyProgressCount={jobsInHealthyProgressCount}");
+
+                // default is healthy
+                var jobsSuccessRate = 1f;
+
+                // if no records exist, instance is healthy
+                if (jobsTotalCount > 0)
+                {
+                    jobsSuccessRate = ((float)(jobsSuccessCount + jobsInHealthyProgressCount)) / jobsTotalCount;
+                }
+
+                var allCalls = this.mediaServiceCallHistoryStorageService.ListByMediaServiceAccountNameAsync(mediaServiceInstanceHealthModel.MediaServiceAccountName, this.timeWindowToLoadMediaServiceCallsInMinutes).GetAwaiter().GetResult().ToList();
+
+                var callsTotalCount = allCalls.Count;
+                var callsSuccessCount = allCalls.Count(c => (int)c.HttpStatus < 500);
+
+                logger.LogInformation($"MediaServiceInstanceHealthService::ReEvaluateMediaServicesHealthAsync calls aggregated data: instanceName={mediaServiceInstanceHealthModel.MediaServiceAccountName} callsSuccessCount={callsSuccessCount} callsTotalCount={callsTotalCount}");
 
                 // default is healthy
-                var successRate = 1f;
+                var callsSuccessRate = 1f;
 
                 // if no records exist, instance is healthy
-                if (totalCount > 0)
+                if (callsTotalCount > 0)
                 {
-                    successRate = ((float)(successCount + inHealthyProgressCount)) / totalCount;
+                    callsSuccessRate = ((float)callsSuccessCount) / callsTotalCount;
                 }
 
+                var successRate = (jobsSuccessRate + callsSuccessRate) / 2;
```

Comment:
Instead of averaging these, should you evaluate them independently and take the worst state?  Meaning if the job success rate is healthy but the API call rate is degraded then mark it as degraded.

Reply:
sounds good, makes sense.